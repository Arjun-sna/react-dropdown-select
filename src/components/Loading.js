import React from 'react';
import styled from '@emotion/styled';

const Loading = ({ parentProps }) =>
  parentProps.loadingRenderer ? (
    parentProps.loadingRenderer()
  ) : (
    <LoadingComponent className="react-dropdown-select-loading" />
  );

Loading.propTypes = {};

const LoadingComponent = styled.div`
  @keyframes dual-ring-spin {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(180deg);
    }
  }

  padding: 0 5px;
  display: block;
  width: auto;
  height: auto;

  :after {
    content: ' ';
    display: block;
    width: 16px;
    height: 16px;
    border-radius: 50%;
    border-width: 1px;
    border-style: solid;
    border-color: deepskyblue transparent;
    animation: dual-ring-spin 0.7s ease-in-out infinite;
    margin: 0 0 0 -10px;
  }
`;

export default Loading;
