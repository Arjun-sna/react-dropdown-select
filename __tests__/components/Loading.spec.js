import React from 'react';
import renderer from 'react-test-renderer';

import Loading from '../../src/components/Loading';

const props = {
  parentProps: {
    loadingRenderer: null
  }
};

it('<Loading/> renders correctly', () => {
  const tree = renderer.create(<Loading {...props}/>).toJSON();

  expect(tree).toMatchSnapshot();
});
