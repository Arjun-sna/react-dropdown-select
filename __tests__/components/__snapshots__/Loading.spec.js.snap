// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading/> renders correctly 1`] = `
<div
  className="react-dropdown-select-loading css-k8dcp9-LoadingComponent e1l5cpc30"
/>
`;
